sycl/source/detail/kernel_program_cache.hpp

```diff
@@ -67,17 +67,16 @@ class KernelProgramCache {
   using PiProgramT = std::remove_pointer<RT::PiProgram>::type;
   using PiProgramPtrT = std::atomic<PiProgramT *>;
   using ProgramWithBuildStateT = BuildResult<PiProgramT>;
-  using ProgramCacheKeyT =
-      std::pair<std::pair<SerializedObj, KernelSetId>, RT::PiDevice>;
+  using ProgramCacheKeyT = std::pair<std::pair<SerializedObj, KernelSetId>,
```

Comment:
Nit: with the type alias for the program map, this could be also simplified to MapType::key_type